### ISSUE TITLE
disable sharing when there is nothing to share

### ISSUE THREAD

**Issue by DuckSoft** [lhy0403/Qv2ray#165]
reproduction:
1. clean install and launch
2. bug occurs: you can share even no host is imported
![图片](https://user-images.githubusercontent.com/7822648/70279100-a53c3e00-17f0-11ea-9dca-ad1841f60f2b.png)

**Comment by lhy0403** [comment]
Marked as done, will close soon after commit.

**Comment by DuckSoft** [comment]
Well done! お疲れ様でした。

---
ps: @lhy0403 
when you want to use `close`, `fix` or some other commit message instructions to **close multiple issues** at a time, **repeating the verb itself is a must**. Look at the following picture:
![image](https://user-images.githubusercontent.com/7822648/70325479-07d31f80-186d-11ea-88ca-1d0d9f14d15a.png)
this picture shows that your commit message is intended to show that it shall **fix** #165, #164, #163, #161 and #160. But in the picture, the `fixed` verb only affects the issue right after it, #165. As for issue #164, #163, #161 and #160, they are shown as 'mentioned in commit' instead of 'can be fixed by commit', shown as follows:
![image](https://user-images.githubusercontent.com/7822648/70325677-80d27700-186d-11ea-8e7b-0ad7ece6bbb1.png)
a correct example of commit message of this use case can be:
```
[fix] several security patches.

this commit fixes #165, fixes #163, fixes #161 and fixes #16 
```
fyi.
